stablehlo/reference/Ops.cpp

```diff
@@ -319,6 +335,35 @@ Tensor evalAndOp(const Tensor &lhs, const Tensor &rhs, TensorType resultType) {
   return result;
 }
 
+Tensor evalBatchNormInferenceOp(const Tensor &operand, const Tensor &scale,
+                                const Tensor &offset, const Tensor &mean,
+                                const Tensor &variance, APFloat epsilon,
+                                Axis featureIndex, TensorType resultType) {
+  auto scaleBroadcast =
+      evalBroadcastInDimOp(scale, {featureIndex}, operand.getType());
+  auto offsetBroadcast =
+      evalBroadcastInDimOp(offset, {featureIndex}, operand.getType());
+  auto meanBroadcast =
+      evalBroadcastInDimOp(mean, {featureIndex}, operand.getType());
+  auto varianceBroadcast =
+      evalBroadcastInDimOp(variance, {featureIndex}, operand.getType());
+  auto epsilonBroadcast = evalBroadcastInDimOp(
+      makeTensor(DenseElementsAttr::get(
+          RankedTensorType::get({}, getF32Type(resultType.getContext())),
+          {epsilon})),
+      {}, operand.getType());
+
+  auto centeredOperand = evalSubtractOp(operand, meanBroadcast, resultType);
```

Comment:
for the eval functions here and below, I do not think we should use the `resultType` as the last argument,  but use the `getType()`  of any operand, say the first. 

I agree that result type of any of these evals must be equal to `resultType`, but that is an information we derived from the type inference of a series of associated operations flowing to the result.

Reply:
Agreed.